services/blockchain-indexer/shared/dataService/business/events.js

```diff
@@ -224,10 +248,10 @@ const getEvents = async params => {
 				event = eventsFromCache.find(entry => entry.index === index);
 			}
 
-			const [{ id, timestamp } = {}] = await blocksTable.find({ height, limit: 1 }, [
-				'id',
-				'timestamp',
-			]);
+			if (!blocksCache[height]) {
+				blocksCache[height] = await blocksTable.find({ height, limit: 1 }, ['id', 'timestamp']);
+			}
+			const [{ id, timestamp } = {}] = blocksCache[height];
```

Comment:
Why do we need this? As height is indexed, this should be cheap operation. Also, this can produce inconsistent results when fork happens for a height.

Reply:
It was added last night when we were debugging and optimizing the code. But it's also fine if we remove this caching.